pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -530,9 +625,74 @@ pub mod pallet {
 			domain_address: DomainAddress,
 			message: T::Message,
 		) -> (DispatchResult, Weight) {
-			let weight = Weight::from_parts(0, T::Message::max_encoded_len() as u64)
+			let mut weight = Weight::from_parts(0, T::Message::max_encoded_len() as u64)
 				.saturating_add(LP_DEFENSIVE_WEIGHT);
 
+			let routers_count = match DomainMultiRouters::<T>::get(domain_address.domain()) {
+				Some(r) => r.len(),
+				None => return (Err(Error::<T>::MultiRouterNotFound.into()), weight),
+			};
+
+			weight = weight.saturating_add(T::DbWeight::get().reads(1));
+
+			if routers_count <= 1 {
+				return (Err(Error::<T>::NotEnoughRouters.into()), weight);
+			}
+
+			let expected_proof_count: u32 = (routers_count - 1) as u32;
+
+			let (message_proof, message_proof_count) = match message.get_message_proof() {
+				None => {
+					let message_proof = message
+						.to_message_proof()
+						.get_message_proof()
+						.expect("message proof ensured by 'to_message_proof'");
+
+					InboundMessages::<T>::insert(message_proof, message.clone());
```

Comment:
This has to be changed in order to support the same message being sent more than once.

Reply:
Done.